docs/guides/connecting_to_your_data/overview.md

```diff
@@ -0,0 +1,216 @@
+- In filesystems, an `InferredAssetDataConnector` infers the `data_asset_name` by using a regex that takes advantage of patterns that exist in the filename or folder structure.  If your source data system is designed so that it can easily be parsed by regex, this will allow new data to be included by the Datasource automatically.  The `InferredAssetSqlDataConnector` provides similar functionality for SQL based source data systems.
+- A `ConfiguredAssetDataConnector`, which allows you to have the most fine-tuning by requiring an explicit listing of each Data Asset you want to connect to.
+- A `RuntimeDataConnector` which enables you to use a `RuntimeBatchRequest` to wrap either an in-memory dataframe, filepath, or SQL query.
+
+In the `data_connectors` dictionary you may define multiple Data Connectors, including different types of Data Connectors, so long as they all have unique values in the place of the `<name_of_your_data_connector>` key.  We provide detailed guidance to help you decide on which Data Connectors to use in our guide: [How to choose which DataConnector to use](/docs/guides/connecting_to_your_data/how_to_choose_which_dataconnector_to_use).
+
+The `<additional_keys_based_on_source_data_system>` will be things like `batch_identifiers`, `base_directory`, and `default_regex` for filesystems, or  `batch_identifiers` for SQL based data systems. For specifics on the additional keys that you can use in your Data Connectors' configurations, please see the corresponding guide for connecting to a specific source data system (since the keys you will need to define will depend on the source data system you are connecting to).
+
+### 3. Test your configuration
+
+Because the configurations for Datasources can vary depending on the underlying data system they are connecting to, Great Expectations provides a convenience function that will help you determine if there are errors in your configuration.  This function is `test_yaml_config()`.  Using `test_yamal_config` in a Jupyter Notebook is our recommended method for testing Datasource configuration.  Of course, you can always edit and test YAML configs manually, and instantiate Datasources through code.  When executed, `test_yamal_config()` will instantiate the component and run through a self check procedure to verify the component works as expected.
+
+In the case of a Datasource, this means:
+- confirming that the connection works.
+- gathering a list of available DataAssets (e.g. tables in SQL; files or folders in a filesystem)
+- verifying that it can successfully fetch at least one Batch from the source.
+
+If something about your configuration wasn't set up correctly, `test_yamal_config()` will raise an error.  Whenever possible, `test_yamal_config()` provides helpful warnings and error messages.  It can't solve every problem, but it can solve many.
```

Comment:
```suggestion
If something about your configuration wasn't set up correctly, `test_yaml_config()` will raise an error.  Whenever possible, `test_yaml_config()` provides helpful warnings and error messages.  It can't solve every problem, but it can solve many.
```